backend/api/endpoints/game.py

```diff
@@ -1,28 +1,36 @@
-from typing import Optional
-
 from fastapi import APIRouter, Response, Cookie
+from typing import Optional
 
+from db_funcs import *
 from api.logic import user_management
-from db_funcs.cipherGameFuncs import *
 
-router = APIRouter()
+@router.get('/api/ciphers/{cipher_game_id}/show/{cipher_id}')
+def get_cipher_info(cipher_game_id: int, cipher_id: int, response: Response, session_cookie: Optional[str] = Cookie(None)) -> Optional[Cipher]:
+    """
+        Getting informations about specified cipher
+
+        :param cipher_game_id: Game id to use
+        :param cipher_id: Cipher id to use
+        :return 200 everything is OK
+                400 cipher wasn't found
+                401 no permission for getting information about cipher
+    """
 
-@router.get("/api/games")
-def get_all_games(response: Response, session_cookie: Optional[str] = Cookie(None)):
     user = user_management.get_user_by_token(session_cookie)
+
     if user is None:
         response.status_code = 401
         return None
-    if user.is_root:
-        games = get_all_cipher_games()
-    else:
-        # Get all games, that are visible to public at the moment,
-        # or user is admin of the given game.
-        games = get_visible_games(user.person_id)
-    if games is None:
+
+    cipher_info = cipherGameFuncs.get_cipher_info(cipher_game_id, cipher_id)
+    if cipher_info is None:
         response.status_code = 400
         return None
-    else:
-        response.status_code = 200
-    return games
 
+    if not is_visible(cipher_id) and not user.is_root and not is_staff(cipher_game_id, user.person_id):
```

Comment:
Jestli jsem to nepřehlédl, tak to vypadá, že chybí kontrola, že tým ve kterém je daný hráč vyřešil šifru, která je daná jako prerekvizita (jestli nějaká je)